smartcontract/programs/doublezero-serviceability/src/state/interface.rs

```diff
@@ -192,27 +191,228 @@ impl Default for InterfaceV1 {
+            InterfaceCYOA::GREOverCable => write!(f, "GREOverCable"),
+        }
+    }
+}
+
+#[repr(u8)]
+#[derive(BorshSerialize, BorshDeserialize, Debug, Copy, Clone, PartialEq, Default)]
+#[borsh(use_discriminant = true)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum InterfaceDIA {
+    #[default]
+    None = 0,
+    DIA = 1,
+}
+
+impl From<u8> for InterfaceDIA {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => InterfaceDIA::DIA,
+            _ => InterfaceDIA::None,
+        }
+    }
+}
+impl fmt::Display for InterfaceDIA {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            InterfaceDIA::None => write!(f, "none"),
+            InterfaceDIA::DIA => write!(f, "yes"),
+        }
+    }
+}
+
+#[repr(u8)]
+#[derive(BorshSerialize, BorshDeserialize, Debug, Copy, Clone, PartialEq, Default)]
+#[borsh(use_discriminant = true)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum RoutingMode {
+    #[default]
+    Static = 0,
+    BGP = 1,
+}
+
+impl From<u8> for RoutingMode {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => RoutingMode::Static,
+            1 => RoutingMode::BGP,
+            _ => RoutingMode::Static,
+        }
+    }
+}
+
+impl fmt::Display for RoutingMode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            RoutingMode::Static => write!(f, "static"),
+            RoutingMode::BGP => write!(f, "bgp"),
+        }
+    }
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct InterfaceV2 {
+    pub status: InterfaceStatus,       // 1
+    pub name: String,                  // 4 + len
+    pub interface_type: InterfaceType, // 1
+    pub interface_cyoa: InterfaceCYOA, // 1
+    pub interface_dia: InterfaceDIA,   // 1
+    pub loopback_type: LoopbackType,   // 1
+    pub bandwidth: u64,                // 8
+    pub cir: u64,                      // 8
+    pub mtu: u16,                      // 2
+    pub routing_mode: RoutingMode,     // 1
+    pub vlan_id: u16,                  // 2
+    pub ip_net: NetworkV4,             // 4 IPv4 address + 1 subnet mask
+    pub node_segment_idx: u16,         // 2
+    pub user_tunnel_endpoint: bool,    // 1
+}
+
+impl InterfaceV2 {
+    pub fn size(&self) -> usize {
+        Self::size_given_name_len(self.name.len())
+    }
+
+    pub fn to_interface(&self) -> Interface {
+        Interface::V2(self.clone())
+    }
+
+    pub fn size_given_name_len(name_len: usize) -> usize {
+        1 + 4 + name_len + 1 + 1 + 1 + 1 + 8 + 8 + 2 + 1 + 2 + 5 + 2 + 1
+    }
+}
+
+impl TryFrom<&[u8]> for InterfaceV2 {
+    type Error = ProgramError;
+
+    fn try_from(mut data: &[u8]) -> Result<Self, Self::Error> {
```

Comment:
This could use `BorshDeserializeIncremental` instead of defining the `TryFrom<&[u8]>` manually